### ISSUE TITLE
Simulator crashes when displaying shield in maneuver

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#1652]
During turn-by-turn navigation in CarPlay, when we go to update the maneuver to display a visual instruction that includes a shield, the application crashes just after calling `CarPlayNavigationViewController.updateManeuvers(for:)`:

```
#0	0x000000010c4d1071 in CarPlayNavigationViewController.visualInstructionDidChange(_:) at /path/to/mapbox-navigation-ios/MapboxNavigation/CarPlayNavigationViewController.swift:203
#1	0x000000010c4d135c in @objc CarPlayNavigationViewController.visualInstructionDidChange(_:) ()
```
https://github.com/mapbox/mapbox-navigation-ios/blob/76ccc5813665c99f74ce6f33a89cc474d940c3d0/MapboxNavigation/CarPlayNavigationViewController.swift#L203

Here’s the stack trace:

```
#0	0x0000000110ce7705 in objc_exception_throw ()
#1	0x0000000112a9d0f5 in +[NSException raise:format:] ()
#2	0x000000011162898c in __47-[CPInterfaceController _connectionInterrupted]_block_invoke ()
```

Based on the assembly code for that block, the exception is `Connection to host was interrupted`, indicating that the simulator has terminated its connection to the application.

The issue only occurs when the instruction’s attributed string contains a ShieldAttachment.

/cc @bsudekum @JThramer @frederoni

**Comment by bsudekum** [comment]
Changing each custom subclassed NSAttributedText init method in this file, example:

https://github.com/mapbox/mapbox-navigation-ios/blob/f185b563916631500f534d0fbf16daab8adfc6a1/MapboxNavigation/InstructionPresenter.swift#L200

to use the normal NSAttributedText init method fixes the issue.

**Comment by JThramer** [comment]
@bsudekum, `GenericShieldAttachment` is a subclass of `NSTextAttachment`, not `NSAttributedText.`

**Comment by bsudekum** [comment]
Ah yeah, my mistake.

**Comment by akitchen** [comment]
Please see page 29 of the developer docs - what you're trying to do is probably not supported

**Comment by 1ec5** [comment]
Yes, the code is currently using `attributedInstructionVariants`. Given that Apple Maps doesn’t display shields inline with the instruction text as Google Maps does, I think it’s likely that text attachments within attributed strings are somewhat of a wild west in CarPlay.

**Comment by 1ec5** [comment]
> Changing each custom subclassed NSAttributedText init method in this file to use the normal NSAttributedText init method fixes the issue.

While that works around the crash, unfortunately the image doesn’t show up in the simulator, nor does the textual alternative.

**Comment by 1ec5** [comment]
Speculative fix in #1665.

**Comment by 1ec5** [comment]
> the shields still don’t show up in the simulator

→ #1867
